src/SFML/System/Win32/LibraryLoader.hpp

```diff
@@ -0,0 +1,120 @@
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_LIBRARY_LOADERWIN32_HPP
+#define SFML_LIBRARY_LOADERWIN32_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <SFML/System/NonCopyable.hpp>
+#include <SFML/System/String.hpp>
+#include <windows.h>
+
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+/// \brief todo
+///
+////////////////////////////////////////////////////////////
+class SFML_SYSTEM_API LibraryLoader : public NonCopyable
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// todo
+    ///
+    ////////////////////////////////////////////////////////////
+    LibraryLoader() = delete;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// todo
+    ///
+    /// \param libraryName todo
+    ///
+    ////////////////////////////////////////////////////////////
+    LibraryLoader(const sf::String& libraryName);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default destructor
+    ///
+    /// todo
+    ///
+    ////////////////////////////////////////////////////////////
+    virtual ~LibraryLoader();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// \return todo
+    ///
+    ////////////////////////////////////////////////////////////
+    bool getIsLoaded();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// todo
+    ///
+    /// \param procName todo
+    ///
+    /// \return True if the TODO
+    ///
+    ////////////////////////////////////////////////////////////
+    template<typename TFuncType>
+    bool getProcAddress(TFuncType& procAddress, const sf::String& procName);
+
+private:
+    
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// todo
+    ///
+    ////////////////////////////////////////////////////////////
+    void freeLibrary();
+
+    ////////////////////////////////////////////////////////////
+    // Member data
+    ////////////////////////////////////////////////////////////
+    sf::String m_libraryName;           //!< todo
+    HMODULE    m_moduleHandle{nullptr}; //!< todo
```

Comment:
Is the `{nullptr}` initialization needed at all? The constructor always seems to initialize `m_moduleHandle`.